src/worker/worker.c

```diff
@@ -263,6 +288,14 @@ static void kill_job(struct nm_event_execution_properties *event)
 		(void)kill(-cp->ei->pid, SIGKILL);
 		return;
 	}
+	/* check if the child we'r killing belongs to this worker process */
+	wpid = getpid();
+	get_process_parent_id(pid, &ppid);
+	if (-1 != ppid && ppid != wpid) {
```

Comment:
If we can't find the process, we proceed to sending a SIGKILL further down below?
I think this should be `if (ppid == -1 || ppid != wpid)`.

Reply:
I think it's the os (operating system ) issue if we can't find a running process in os proc list.
Another case (process just finished and we din't find in proc list).
In my opinion, though it's rare case but in both cases going down may not be harmful.

Reply:
A third case is where a process spawns right after we check for its existence. Given it's a slim chance, but it could happen if we're spawning a whole lot of processes, which we are in a large naemon system. Then we'll again be sending SIGKILL to a process that isn't the workers child.

Reply:
Then I think it's ok to modify this `if ... case` .